xapian-core/api/valuerangeproc.cc

```diff
@@ -439,4 +439,100 @@ NumberRangeProcessor::operator()(const string& b, const string& e)
     return Xapian::Query(Xapian::Query::OP_INVALID);
 }
 
+static const int size_map[26] = {
+    0,
+    1, // B
+    0, 0, 0, 0,
+    1024 * 1024 * 1024, // G
+    0, 0, 0,
+    1024, // K
+    0,
+    1024 * 1024, // M
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
+};
+
+static int
+get_bytes(char ch)
+{
+	// It takes character as input and return number of bytes
+	// B - 1 byte
+	// K - 1024 bytes
+	// M - 1024*1024 bytes
+	// G - 1024*1024*1024 bytes
+	if (ch >= 'A' && ch <= 'Z' && size_map[ch - 'A'] > 0) {
+    return size_map[ch - 'A'];
+	} else {
+	return 0;
+	}
+}
+
+Xapian::Query
+FileSizeRangeProcessor::operator()(const string& b, const string& e){
+	// Here b and e will be like "100K" and "1M"
+	double size_b, size_e;
+	char unit_b = '\0', unit_e = '\0';
+	string temp_b = b, temp_e = e;
+
+	if (!b.empty()) {
+		errno = 0;
+		char b_back = b.back();
+		if (b_back =='B' || b_back =='K' || b_back =='M' || b_back =='G') {
+			// If suffix of b is 'B','K','M','G'
+			unit_b = b_back;
+			temp_b.pop_back();
+		} else if (!isdigit(b_back)) {
```

Comment:
As Olly previously said:

> Use the Xapian-specific `C_isdigit()` instead of `isdigit()` in Xapian code (also true for `isalpha()`, and other functions in the same family).
>
> The set of characters for which `isdigit()` returns a true value may vary if `setlocale()` has been called, which we want to avoid.
>
> Also `isdigit()` isn't defined for negative values, which is a problem on platforms where `char` is `signed char`,  Our `C_isdigit()` function works for both signed and unsigned char (the GNU libc implementation also does, but we can't rely on the system's libc extending the range of these functions for us, and it's cumbersome to have to write `isdigit(static_cast<unsigned char>(ch))` every time).